### DISCUSSION TITLE
Osu "Workshop" for game mods/additions?

### DISCUSSION THREAD

**Discussion by ghost** [ppy/osu#22759] [Ideas] [answered]
Hello, just curious about the feasibility of an osu workshop, similar to the workshops for Steam games? Would it be too complex or intensive to implement? The mods/additions would still have to be approved the same way additions currently are on Github, so no chance for malicious modifications, cheating, bad code etc. I think it could help to simplify existing interface customization for players, and I also think it could open up a lot of possibilities for creative projects, as people could work on optional features that wouldn't go into the main client.

**Comment by bdach** [comment, marked as answer]
A custom ruleset listing is planned in the far future as part of https://github.com/ppy/osu/issues/11736. Not sure what else you'd want to include in this "workshop".

**Reply by ghost** [reply]
As a few examples, it could include things people have wanted to add, like story mode or keypress overlay for streams, and useful tools that already exist but as outside applications, such as something like osu-tracker and miss analyzer. Osu has a very passionate community and a lot of projects are started, such as osu!skills, the osu!alt scores analyzer, and OMDB. If there were an avenue for people to implement their ideas in-client as "DLC", in a curated place for add-ons similar to Steam Workshop, I think a lot of neat things could come of it.

**Reply by bdach** [reply]
There is no real reason not to just PR most of what you mention to this repository. Maybe a workshop type thing for those kinds of extensions could happen, but for now I don't really see a real need.

**Reply by peppy** [reply]
Rulesets are the only method of extensions we have right now. Anything further will not be supported in the foreseeable future.
